connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointTask.java

```diff
@@ -306,7 +308,16 @@ Map<String, Map<TopicPartition, OffsetAndMetadata>> syncGroupOffset() {
 
     void syncGroupOffset(String consumerGroupId, Map<TopicPartition, OffsetAndMetadata> offsetToSync) {
         if (targetAdminClient != null) {
-            targetAdminClient.alterConsumerGroupOffsets(consumerGroupId, offsetToSync);
+            AlterConsumerGroupOffsetsResult result = targetAdminClient.alterConsumerGroupOffsets(consumerGroupId, offsetToSync);
+            result.all().whenComplete((v, throwable) -> {
+                if (throwable != null) {
+                    if (throwable.getCause() instanceof UnknownMemberIdException) {
```

Comment:
How stable is the API for the exceptions we'll see here? Can we be reasonably certain that the exception we want to examine will always be wrapped?

Reply:
The outer exception depends how the future is completed but I _think_ the actual error should always be wrapped.

Reply:
I took a closer look at the [`KafkaFuture` Javadocs](https://kafka.apache.org/32/javadoc/org/apache/kafka/common/KafkaFuture.html).

The docs for `KafkaFuture::whenComplete` state (emphasis mine):

> Returns a new KafkaFuture **with the same result or exception as this future**, that executes the given action when this future completes. When this future is done, **the given action is invoked with the result (or null if none) and the exception (or null if none) of this future as arguments**."

Given this, we know that `whenComplete` receives the same exception that would be thrown by, e.g., `KafkaFuture::get`, if the action failed.

The method signatures for `KafkaFuture` declare that they throw `ExecutionException` if the action fails, which always wraps the cause of failure.

So, I'm convinced that we can expect this wrapping 👍